### DISCUSSION TITLE
Legality of using Qt to build pgmodeler 0.94 from Source.

### DISCUSSION THREAD

**Discussion by PowerUser1234** [pgmodeler/pgmodeler#1617] [General]
-According to the licensing that applies to the pgmodeler 0.9.4 source code resources, it is legal to build and use pgmodeler
in a commercial and non commercial setting, so long as the software itself is not sold.

-Is this changed or rendered not the case bcause the Qt library has to be utilised to build 0.9.4?  Does using Qt to build pgmodeler 0.9.4 adjust or reduce the scope of the licensing to thereby compiled and build pgmodeler program, on any operating system?

**Comment by rkhaotix** [comment]
> -According to the licensing that applies to the pgmodeler 0.9.4 source code resources, it is legal to build and use pgmodeler
> in a commercial and non commercial setting, so long as the software itself is not sold.

You can sold the software, but you can't close its source, unless you use the commercial Qt license. 



> -Is this changed or rendered not the case bcause the Qt library has to be utilised to build 0.9.4? Does using Qt to build pgmodeler 0.9.4 adjust or reduce the scope of the licensing to thereby compiled and build pgmodeler program, on any operating system?

Could you please elaborate a bit more? Do you want to use pgModeler in a more restricted manner?

**Comment by PowerUser1234** [comment]
1) The way that the Qt license works on its own is that Qt is free for free, open source software (FOSS) in a non commercial context.
    That changes in a commercial context, and people legally have to pay Qt.

2) How does 1) Carry forward to pgmodeler, particularly if using a source code build of pgmodeler, in a commercial context?  I don't plan to sell pgmodeler. But I don't understand how the Qt license influences pgmodeler, since Qt has to be used to compile and build pgmodeler.
What difference does the Qt license make to a source code build version of pgmodeler?

3)  I have another important question, about using pgmodeler's code snippet feature.  Can someone in the know please read and
respond to [https://github.com/pgmodeler/pgmodeler/issues/1602](https://github.com/pgmodeler/pgmodeler/issues/1602) ?

**Reply by rkhaotix** [reply]
> 1. The way that the Qt license works on its own is that Qt is free for free, open source software (FOSS) in a non commercial context. That changes in a commercial context, and people legally have to pay Qt.

Not necessarily. People have to pay for Qt if they have the intention to close the source of the application and make it purely commercial, AFAIK.




> 2\. How does 1) Carry forward to pgmodeler, particularly if using a source code build of pgmodeler, in a commercial context?  I don't plan to sell pgmodeler. But I don't understand how the Qt license influences pgmodeler, since Qt has to be used to compile and build pgmodeler. What difference does the Qt license make to a source code build version of pgmodeler?

I think there's no difference or influence regarding Qt license when building pgModeler from source. The main obligation you have when building an open source software using Qt is never close its source otherwise you have to move to commercial license of the framework.

I don't know if I made it clear. Sometimes the licenses subjects may be a lot confusing. Feel free to ask anything else.
